public/app/features/browse-dashboards/components/NewProvisionedFolderForm.tsx

```diff
@@ -110,8 +110,7 @@ function FormContent({ initialValues, repository, workflowOptions, folder, onDis
     }
     const basePath = folder?.metadata?.annotations?.[AnnoKeySourcePath] ?? '';
 
-    // Convert folder title to filename format (lowercase, replace spaces with hyphens)
-    const titleInFilenameFormat = formatFolderName(title); // TODO: this is currently not working, issue created https://github.com/grafana/git-ui-sync-project/issues/314
+    const titleInFilenameFormat = formatFolderName(title);
```

Comment:
I would not mess with the folder name, I will throw a validation error in the form so that they put the name they really want. In that way we will also be able to remove that message "folder will be created with X name". This should simplify the UI, experience and code.